ethereumj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -319,8 +320,11 @@ else if (!currentValue.isZero() && newValue.isZero()) {
                     break;
                 case SHA3:
                     gasCost = gasCosts.getSHA3() + calcMemGas(gasCosts, oldMemSize, memNeeded(stack.peek(), stack.get(stack.size() - 2)), 0);
-                    DataWord size = stack.get(stack.size() - 2);
-                    long chunkUsed = (size.longValueSafe() + 31) / 32;
+                    long size = stack.get(stack.size() - 2).longValueSafe();
+                    if (size == Long.MAX_VALUE) {
```

Comment:
What is this check for?

Reply:
This check is similar to check of [mem gas calculation](https://github.com/ethereum/ethereumj/blob/develop/ethereumj-core/src/main/java/org/ethereum/vm/VM.java#L145). And looks like it's already done in a call to `calcMemGas` prior to this check. I'll remove it in favor of code readability.